diff -burN InsightToolkit-4.13.1.orig/Modules/Core/Common/include/itkUnaryFunctorImageFilter.hxx  InsightToolkit-4.13.1/Modules/Core/Common/include/itkUnaryFunctorImageFilter.hxx
--- InsightToolkit-4.13.1.orig/Modules/Core/Common/include/itkUnaryFunctorImageFilter.hxx	2020-02-17 16:38:42.806358579 +0100
+++ InsightToolkit-4.13.1/Modules/Core/Common/include/itkUnaryFunctorImageFilter.hxx	2020-02-17 16:50:19.622351934 +0100
@@ -61,6 +61,8 @@
     return;
     }

+
+  outputPtr->CopyInformation(inputPtr);
   // Set the output image largest possible region.  Use a RegionCopier
   // so that the input and output images can be different dimensions.
   OutputImageRegionType outputLargestPossibleRegion;