diff -burN pugixml-1.5/src/pugiconfig.hpp pugixml-1.5-modified-v1/src/pugiconfig.hpp
--- pugixml-1.5/src/pugiconfig.hpp	2014-11-25 05:17:23.000000000 +0100
+++ pugixml-1.5-modified-v1/src/pugiconfig.hpp	2016-06-28 15:28:59.000000000 +0200
@@ -35,7 +35,10 @@
 // Tune these constants to adjust memory-related behavior
 // #define PUGIXML_MEMORY_PAGE_SIZE 32768
 // #define PUGIXML_MEMORY_OUTPUT_STACK 10240
+
+// MACCS-Cots 4.3 : modification.
 // #define PUGIXML_MEMORY_XPATH_PAGE_SIZE 4096
+#define PUGIXML_MEMORY_XPATH_PAGE_SIZE 16384
 
 // Uncomment this to switch to header-only version
 // #define PUGIXML_HEADER_ONLY
