diff -burN InsightToolkit-4.7.2.orig/Modules/Core/Common/include/itkVectorImage.h InsightToolkit-4.7.2/Modules/Core/Common/include/itkVectorImage.h
--- InsightToolkit-4.7.2.orig/Modules/Core/Common/include/itkVectorImage.h	2015-05-01 00:43:19.000000000 +0200
+++ InsightToolkit-4.7.2/Modules/Core/Common/include/itkVectorImage.h	2017-12-13 15:07:28.000000000 +0100
@@ -215,7 +215,9 @@
    * allocated yet. */
   void SetPixel(const IndexType & index, const PixelType & value)
   {
-    OffsetValueType offset = m_VectorLength * this->ComputeOffset(index);
+// MACCS 4.3 - optimization
+//    OffsetValueType offset = m_VectorLength * this->ComputeOffset(index);
+    OffsetValueType offset = m_VectorLength * this->DoComputeOffset(index);
 
     for ( VectorLengthType i = 0; i < m_VectorLength; i++ )
       {
@@ -230,7 +232,9 @@
    * pixel on the stack. */
   const PixelType GetPixel(const IndexType & index) const
   {
-    OffsetValueType offset = m_VectorLength * this->ComputeOffset(index);
+// MACCS 4.3 - optimization
+//    OffsetValueType offset = m_VectorLength * this->ComputeOffset(index);
+    OffsetValueType offset = m_VectorLength * this->DoComputeOffset(index);
 
     // Do not create a local for this method, to use return value
     // optimization.
@@ -248,7 +252,9 @@
    * image has actually been allocated yet. */
   PixelType  GetPixel(const IndexType & index)
   {
-    OffsetValueType offset = m_VectorLength * this->ComputeOffset(index);
+// MACCS 4.3 - optimization
+//    OffsetValueType offset = m_VectorLength * this->ComputeOffset(index);
+    OffsetValueType offset = m_VectorLength * this->DoComputeOffset(index);
 
     // Correctness of this method relies of return value optimization, do
     // not create a local for the value.
