diff -burN OTB-7.0.0.orig/Modules/Filtering/ImageManipulation/include/otbStreamingResampleImageFilter.hxx OTB-7.0.0/Modules/Filtering/ImageManipulation/include/otbStreamingResampleImageFilter.hxx
--- OTB-7.0.0.orig/Modules/Filtering/ImageManipulation/include/otbStreamingResampleImageFilter.hxx	2020-03-10 11:26:16.037865195 +0100
+++ OTB-7.0.0/Modules/Filtering/ImageManipulation/include/otbStreamingResampleImageFilter.hxx	2020-03-10 14:52:18.722171735 +0100
@@ -22,7 +22,6 @@
 #define otbStreamingResampleImageFilter_hxx

 #include "otbStreamingResampleImageFilter.h"
-#include "itkProgressAccumulator.h"
 #include "otbImage.h"

 namespace otb
@@ -46,11 +45,6 @@
 template <class TInputImage, class TOutputImage, class TInterpolatorPrecisionType>
 void StreamingResampleImageFilter<TInputImage, TOutputImage, TInterpolatorPrecisionType>::GenerateData()
 {
-  // Set up progress reporting
-  typename itk::ProgressAccumulator::Pointer progress = itk::ProgressAccumulator::New();
-  progress->SetMiniPipelineFilter(this);
-  progress->RegisterInternalFilter(m_WarpFilter, 1.f);
-
   m_WarpFilter->GraftOutput(this->GetOutput());
   m_WarpFilter->UpdateOutputData(m_WarpFilter->GetOutput());
   this->GraftOutput(m_WarpFilter->GetOutput());