--- OTB-7.0.0.orig/Modules/Wrappers/ApplicationEngine/src/otbWrapperApplication.cxx	2019-10-29 10:11:39.000000000 +0100
+++ OTB-7.0.0/Modules/Wrappers/ApplicationEngine/src/otbWrapperApplication.cxx	2020-04-04 11:58:59.000000000 +0200
@@ -1587,7 +1587,7 @@
   else if (dynamic_cast<InputImageListParameter*>(param))
   {
     InputImageListParameter* paramDown = dynamic_cast<InputImageListParameter*>(param);
-    return paramDown->GetNthImage(idx);
+    return paramDown->GetNthImageBase(idx);
   }
   else if (dynamic_cast<OutputImageParameter*>(param))
   {
