diff -burN gdal-1.10.1.orig/ogr/ogrsf_frmts/geojson/jsonc/config.h gdal-1.10.1/ogr/ogrsf_frmts/geojson/jsonc/config.h
--- gdal-2.2.1.orig/ogr/ogrsf_frmts/geojson/libjson/config.h	2017-12-22 15:01:19.000000000 +0100
+++ gdal-2.2.1/ogr/ogrsf_frmts/geojson/libjson/config.h	2017-12-22 15:30:39.000000000 +0100
@@ -23,6 +23,9 @@
 #define __attribute__(x) /* DO NOTHING */
 #endif
 
+#if defined (__GNUC__) && !defined(_GNU_SOURCE)
+#define _GNU_SOURCE
+#endif
 /* Define to 1 if you don't have `vprintf' but do have `_doprnt.' */
 /* #undef HAVE_DOPRNT */
 
